assets/assets_identifiers_ws.py

```diff
@@ -1,49 +1,23 @@
-import asyncio
-import pieces_os_client as pos_client
+from pieces_os_client import StreamedIdentifiers
 import websockets
-from pieces.settings import PiecesSettings
-import threading
-
-class AssetsIdentifiersWS:
-    def __new__(cls,*args,**kwargs):
-        if not hasattr(cls, 'instance'):
-            cls.instance = super(AssetsIdentifiersWS, cls).__new__(cls)
-        return cls.instance
-    def __init__(self, on_message_callback=None):
-        self.ws = None
-        self.is_connected = False
-        self.on_message_callback = on_message_callback
-
-        # Create a new event loop
-        self.loop = asyncio.new_event_loop()
-
-        # Run the event loop in a new thread
-        self.thread = threading.Thread(target=self.start_event_loop, args=(self.loop,))
-        self.thread.start()
-
-        # Open the websocket connection
-        asyncio.run_coroutine_threadsafe(self.open_websocket(), self.loop)
-
-    def start_event_loop(self, loop):
-        asyncio.set_event_loop(loop)
-        loop.run_forever()
-
-    async def open_websocket(self):
-        """Opens a websocket connection"""
-        if self.is_connected: # connect only once
-            return
-        self.ws = await websockets.connect(PiecesSettings.ASSETS_IDENTIFIERS_WS_URL)
-        self.is_connected = True
-        try:
-            async for message in self.ws:
-                self.on_message_callback(pos_client.StreamedIdentifiers.from_json(message))
-        except websockets.exceptions.ConnectionClosed:
-            self.is_connected = False
-            
-
-    async def close_websocket_connection(self):
-        """Close the websocket connection."""
-        if self.ws and self.is_connected:
-            await self.ws.close()
-            self.is_connected = False
 
+from pieces.settings import PiecesSettings
+from pieces.base_websocket import BaseWebsocket
+
+class AssetsIdentifiersWS(BaseWebsocket):
```

Comment:
did the indentation change here? we need to make sure that the formatter stays the same otherwise this can get unruly to review

Reply:
Indentation did not change but I removed most of it to a base websocket class to avoid duplicating code because I will need the same base websocket in the user websocket

Reply:
> Indentation did not change but I removed most of it to a base websocket class to avoid duplicating code because I will need the same base websocket in the user websocket

it looks like we went from 4 spaces to 8 spaces indentation here

Reply:
Yeah true did not notice😅😅